telegram/ext/conversationhandler.py

```diff
@@ -354,12 +354,12 @@ def update_state(self, new_state, key):
             if self.persistent:
                 self.persistence.update_conversation(self.name, key, new_state)
 
-    def _trigger_timeout(self, bot, job):
+    def _trigger_timeout(self, dispatcher):
         self.logger.debug('conversation timeout was triggered!')
-        del self.timeout_jobs[job.context.conversation_key]
+        del self.timeout_jobs[dispatcher.job.context.conversation_key]
         handlers = self.states.get(self.TIMEOUT, [])
         for handler in handlers:
-            check = handler.check_update(job.context.update)
+            check = handler.check_update(dispatcher.job.context.update)
             if check is not None and check is not False:
-                handler.handle_update(job.context.update, job.context.dispatcher, check)
-        self.update_state(self.END, job.context.conversation_key)
+                handler.handle_update(dispatcher.job.update, dispatcher.job.dispatcher, check)
```

Comment:
Which means that handle_update should get passed `context.job.context.update, context.dispatcher, check`
and maybe even add `, context=context` not entirely sure about that.